tests/MongoDB.EntityFrameworkCore.FunctionalTests/Mapping/ValueConverterTests.cs

```diff
@@ -1183,6 +1183,50 @@ class GenericPropertyEntity<T>
         public T value { get; set; }
     }
 
+    [Fact]
+    public void Converter_on_key_and_secondary_property_does_not_cause_key_errors()
```

Comment:
Is this test supposed to be using alternate keys?

Reply:
No, this was the original bug that caused me to go check out alternate keys. It temporarily during the model building process looks like an alternate key even though it isn't - just the way EF builds it. The validation was really in the wrong place but now that validation is remove anyway with this feature and I wanted some coverage.